src/feature/RansacFeatureSetMatcher.cpp

```diff
@@ -51,26 +60,82 @@ double RansacFeatureSetMatcher::matchSets(const std::vector<InterestPoint *> &re
     unsigned int iterations = m_adaptive ? 1e17 : ceil(log(1. - m_successProbability)/log(1. - m_inlierProbability * m_inlierProbability));
     
     // Compute possible correspondences based on 1-NN thresholding
-   std::vector< std::pair<InterestPoint *, InterestPoint *> > possibleCorrespondences;
-   for(unsigned int i = 0; i < data.size(); i++){
-	double minCorrespondenceDistance = 1e17;
-	unsigned int minCorrespondenceIndex = 0;
-	for(unsigned int j = 0; j < reference.size(); j++){
-	    double distance = data[i]->getDescriptor()->distance(reference[j]->getDescriptor());
-	    if(distance < minCorrespondenceDistance){
-		minCorrespondenceDistance = distance;
-		minCorrespondenceIndex = j;
-	    }
-	}
-	if(minCorrespondenceDistance < m_distanceThreshold){
-	    possibleCorrespondences.push_back(std::make_pair(data[i], reference[minCorrespondenceIndex]));
-	}
+    std::vector< std::pair<InterestPoint *, InterestPoint *> > possibleCorrespondences;
+    std::vector<IndexedDistance> minDistances;
+
+    if (!m_flann){
+        for(unsigned int i = 0; i < data.size(); i++){
+            double minCorrespondenceDistance = 1e17;
+            unsigned int minCorrespondenceIndex = 0;
+            for(unsigned int j = 0; j < reference.size(); j++){
+                double distance = data[i]->getDescriptor()->distance(reference[j]->getDescriptor());
+                if(distance < minCorrespondenceDistance){
+                    minCorrespondenceDistance = distance;
+                    minCorrespondenceIndex = j;
+                }
+             }
+
+            if(minCorrespondenceDistance < m_distanceThreshold){
+                possibleCorrespondences.push_back(std::make_pair(data[i], reference[minCorrespondenceIndex]));
+                minDistances.push_back(std::make_pair(minDistances.size(), minCorrespondenceDistance));
+            }
+        }
+    }
+    else { /* m_flann */
+        // Number of nearest neighbours to use
+        size_t nn = 1;
+   
+        // Create and fill the flann matrices from the data and reference descriptors
+        std::vector<double> data_descriptors;
+        std::vector<double> reference_descriptors;
+        std::vector<int> indices(data.size()*nn);
```

Comment:
I recommend to do `const size_t data_size = data.size();` first because it's used in multiple places, and also `if (data_size == 0) { return; }` (or similar) because you divide by it later. But maybe this never happens.